sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -351,10 +270,9 @@ private void initializeCryptoClients() {
 
         if (!checkKeyPermissions(this.key.getKeyOps(), KeyOperation.DECRYPT)) {
             return Mono.error(new UnsupportedOperationException(String.format("Decrypt Operation is not allowed for "
-                + "key with id %s", key.getKid())));
+                + "key with id %s", key.getKeyId())));
```

Comment:
`UnsupportedOperationException` should be wrapped with `logger.logAsError()`

Reply:
done